### DISCUSSION TITLE
Using Cockroach DB backend for ferretdb

### DISCUSSION THREAD

**Discussion by trickster** [FerretDB/FerretDB#2462] [General]
I am trying to use cockroachdb backend for ferretdb. Here's the docker compose file,

https://github.com/trickster/ferret-on-cockroach/blob/main/docker-compose.yml

It gives me this error

```

DeprecationWarning: Collection.insert() is deprecated. Use insertOne, insertMany, or bulkWrite.
Uncaught:
MongoBulkWriteError: [msg_insert.go:38 pg.(*Handler).MsgInsert] [pg.go:137 pg.(*Handler).DBPool] pgdb.NewPool: ERROR: database "ferretdb" does not exist (SQLSTATE 3D000)
Result: BulkWriteResult {
  insertedCount: 0,
  matchedCount: 0,
  modifiedCount: 0,
  deletedCount: 0,
  upsertedCount: 0,
  upsertedIds: {},
  insertedIds: { '0': ObjectId("643e0b21a5cd4d0724df4843") }
```

**Comment by AlekSi** [comment]
> https://github.com/trickster/ferret-on-cockroach/blob/main/docker-compose.yml

This link gives me 404.

But I think the problem is the missing database. So please create a "ferretdb" database in the CockroachDB.

**Reply by trickster** [reply]
Oops. I made it public now. Thanks for the suggestion

**Reply by trickster** [reply]
I get this error now

```
ferretdb> db.users.insertOne({ name: 'John Doe', email: 'johndoe@example.com' })
MongoServerError: [msg_insert.go:38 pg.(*Handler).MsgInsert] [pg.go:137 pg.(*Handler).DBPool] pgdb.checkConnection: number of field descriptions must equal number of destinations, got 2 and 3
```

**Reply by AlekSi** [reply]
See https://github.com/cockroachdb/cockroach/issues/101715. I think we could make a workaround…

**Reply by AlekSi** [reply]
https://github.com/FerretDB/FerretDB/pull/2464

**Reply by AlekSi** [reply]
Once [this run](https://github.com/FerretDB/FerretDB/actions/runs/4730645409) finishes, please try with produced images

**Reply by trickster** [reply]
I get new error 

```
ferretdb> db.users.insertOne({ name: 'John Doe', email: 'johndoe@example.com' })
MongoServerError: [msg_insert.go:38 pg.(*Handler).MsgInsert] [pg.go:137 pg.(*Handler).DBPool] pgdb.checkConnection: "lc_collate" is "C.UTF-8"
```

**Reply by AlekSi** [reply]
Please change you locale to `en_US.UTF-8`.
